pkg/adapter/check.go

```diff
@@ -53,3 +54,7 @@ func (r *CheckResult) Combine(otherPtr interface{}) interface{} {
 	}
 	return r
 }
+
+func (r *CheckResult) String() string {
+	return fmt.Sprintf("CheckResult: status:%d, duration:%d, usecount:%d", r.Status, r.ValidDuration, r.ValidDuration)
```

Comment:
r.Status should have a non number version too. S_OK etc.

Reply:
Shall we also print the message inside r ?

Reply:
D.